packages/protocol/contracts/common/Accounts.sol

```diff
@@ -201,14 +202,12 @@ contract Accounts is
    *      is 0x0 or msg.sender).
    */
   function setWalletAddress(address walletAddress, uint8 v, bytes32 r, bytes32 s) public {
-    require(isAccount(msg.sender), "Unknown account");
     if (!(walletAddress == msg.sender || walletAddress == address(0x0))) {
-      address signer = Signatures.getSignerOfAddress(msg.sender, v, r, s);
-      require(signer == walletAddress, "Invalid signature");
+      require(isAccount(msg.sender), "Unknown account");
```

Comment:
oh oops yeah didn't mean to move this require into the conditional, will remove instead since it's in `setIndexedSigner` anyways

Reply:
Not exactly sure why this conditional exists but does it make sense for `setWalletAddress` to no-op if it isn't entered?

Reply:
I think the idea before was to have a default/test scenario where the wallet address is allowed to be set to the sender address or `0x0`, and in this scenario we don't need to check if `signer == walletAddress`, but we no longer want to allow this scenario anymore. Would it make more sense to return an error instead of just doing nothing?

Reply:
I would vote for us to just remove the conditional and always attempt to set the wallet signer
the `legacyAuthorizeSignerWithSignature` and `setIndexedSigner` both have sufficient protections within them

Reply:
just breaks a lot of tests cases :(